src/6_branch.js

```diff
@@ -783,32 +783,21 @@ Branch.prototype['first'] = wrap(callback_params.CALLBACK_ERR_DATA, function(don
 /*** +TOC_ITEM #setidentityidentity-callback &.setIdentity()& ^ALL ***/
 Branch.prototype['setIdentity'] = wrap(callback_params.CALLBACK_ERR_DATA, function(done, identity) {
 	var self = this;
-	this._api(
-		resources.profile,
-		{
-			"identity": identity
-		},
-		function(err, data) {
-			if (err) {
-				done(err);
-			}
-
-			data = data || { };
-			self.identity_id = data['identity_id'] ? data['identity_id'].toString() : null;
-			self.sessionLink = data['link'];
-
-			self.identity = identity;
-			data['developer_identity'] = identity;
-
-			data['referring_data_parsed'] = data['referring_data'] ?
-				safejson.parse(data['referring_data']) :
-				null;
+	if (identity) {
+		var data = {
+			identity_id: self.identity_id,
+			session_id: self.session_id,
+			link: self.sessionLink
+		};
+		self.identity = identity;
+		// store the identity
+		session.patch(self._storage, { "identity": identity }, true);
```

Comment:
would this store it to local storage as well?

Reply:
yes